### DISCUSSION TITLE
Group by and include related fields

### DISCUSSION THREAD

**Discussion by andreas83** [prisma/prisma#10890] [Q&A] [answered]
**The Problem:**
I have a time tracking, client and project table and i need a way to get the hours spend on different clients. 

My Model:
```
model Tracking {
  id         Int     @id @default(autoincrement())
  start      Int     @db.UnsignedInt
  end        Int     @db.UnsignedInt
  desc       String?
  isTracking Boolean @default(false)
  isDeleted  Boolean @default(false)
  projectId  Int?
  project    Project?   @relation(fields: [projectId], references: [id])
  userId     Int?
  user       User?   @relation(fields: [userId], references: [id])
}

model Client {
  id        Int       @id @default(autoincrement())
  name      String
  isDeleted Boolean   @default(false)
  userId    Int?
  user      User?     @relation(fields: [userId], references: [id])
  Project   Project[]
}

model Project {
  id        Int     @id @default(autoincrement())
  name      String
  isDeleted Boolean @default(false)
  clientId  Int
  client    Client  @relation(fields: [clientId], references: [id])
  Tracking  Tracking[]
}

```

**My Query:**

```
    const tracking = await prisma.tracking.groupBy({
      by: ['projectId'],
      _sum: {
        start: true,
        end: true,
      },
      where: {isDeleted:false}
     
    } );
```

**Result:** 
`
[
   {
      "projectId":1,
      "_sum":{
         "start":1639400400,
         "end":1639407600
      }
   },
   {
      "projectId":3,
      "_sum":{
         "start":1639393200,
         "end":1639394100
      }
   }
]
`

**The Trouble:**

As soon i try to "select" the project name i get following error:

_Unknown field `project` for select statement on model TrackingGroupByOutputType.  Did you mean `projectId`?_

Also "include" gives following error:
_Please either use `include` or `select`, but not both at the same time._

So to me it looks like there is no way to get some fields from a related field while doing group by.
Funny enough, i was able to do some conditions on the related table but not selecting them :confused: 

Related Tickets 
#7454 and #9289

**Comment by AaravShah042** [comment, marked as answer]
https://github.com/prisma/prisma/issues/16243#issue-1445823500

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
